packages/core/src/component.ts

```diff
@@ -33,6 +42,10 @@ export interface Component {
   servers?: {
     [name: string]: Constructor<Server>;
   };
+  /**
+   * An array of directory to serve static assets
+   */
+  staticAssets?: staticAssetsSettings[];
```

Comment:
I think these REST-specific extensions should be defined in a new interface inside `@loopback/rest`, because applications using different transports (e.g GraphQL or gRPC) cannot support static assets.

Reply:
Should we generalize it to allow components to contribute `middleware` based routes?

Reply:
Sounds like a separate feature to me.

Reply:
+1 to leave middleware-based routes out of scope.

Personally, I would like to avoid middleware-based routes for as long as we can. LB4 already supports handler-based routes that play nice with OpenAPI, I would prefer extensions to use that approach.